docs/strategies/retry.md

```diff
@@ -19,14 +19,14 @@
 new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions());
 
 // For instant retries with no delay
-new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+new ResiliencePipelineBuilder().AddRetry(new()
```

Comment:
Reading the code without IDE support is a fair point. 

In this case what if we do not repeat the `new ResiliencePipelineBuilder().AddRetry(` for each and every case. Something like this (removed comments for brevity):
```cs
var options = new RetryStrategyOptions();
new ResiliencePipelineBuilder()
    .AddRetry(options);

options = new RetryStrategyOptions
{
    Delay = TimeSpan.Zero
};

options = new RetryStrategyOptions
{
    ShouldHandle = new PredicateBuilder().Handle<SomeExceptionType>(),
    BackoffType = DelayBackoffType.Exponential,
    UseJitter = true,  
    MaxRetryAttempts = 4,
    Delay = TimeSpan.FromSeconds(3),
};

options = new RetryStrategyOptions
{
    MaxRetryAttempts = 2,
    DelayGenerator = static args =>
    {
        var delay = args.AttemptNumber switch
        {
            0 => TimeSpan.Zero,
            1 => TimeSpan.FromSeconds(1),
            _ => TimeSpan.FromSeconds(5)
        };

        return new ValueTask<TimeSpan?>(delay);
    }
};

var genericOptions = new RetryStrategyOptions<HttpResponseMessage>
{
    DelayGenerator = static args =>
    {
        if (args.Outcome.Result is HttpResponseMessage responseMessage &&
            TryGetDelay(responseMessage, out TimeSpan delay))
        {
            return new ValueTask<TimeSpan?>(delay);
        }

        return new ValueTask<TimeSpan?>((TimeSpan?)null);
    }
};
new ResiliencePipelineBuilder<HttpResponseMessage>()
    .AddRetry(genericOptions);
...
```

Reply:
Maybe we split it into multiple snippets?

The first one shows using it in the simple case:

```csharp
var options = new RetryStrategyOptions();
new ResiliencePipelineBuilder()
    .AddRetry(options);
```

And then the others we just do similar to this?

```csharp
var options = new RetryStrategyOptions
{
    Delay = TimeSpan.Zero
};
```

Otherwise with the inplace mutation of `options` being used without any further calls, someone might misunderstand and copy paste it and think the later changes will do something.

Alternatively show all the possible ways to assign `options`, and then have the following bit at the end:

```csharp
new ResiliencePipelineBuilder()
    .AddRetry(options);
```

Reply:
[Preview](https://github.com/peter-csala/Polly/tree/update-main-readme#retry): What do you think?

Reply:
Cool, then let me refactor the rest as well